coderedcms/migrations/0020_footerorderable_navbarorderable.py

```diff
@@ -0,0 +1,41 @@
+# Generated by Django 3.1.7 on 2021-03-12 19:53
+
+from django.db import migrations, models
+import django.db.models.deletion
+import modelcluster.fields
+
+
+class Migration(migrations.Migration):
```

Comment:
So what you want to do now is manually select the site's current navbar and footers in the chooser, so that older sites will be converted to the new way of doing it.

For this you can add a `RunPython` operation (the operations are run sequentially) after the other operations are done. This function should loop through the current navbars/footers and create NavbarOrderable and FooterOrderable models which link them up to the LayoutSettings.

See example in the Data Migrations section of django docs: https://docs.djangoproject.com/en/3.1/topics/migrations/#data-migrations

Reply:
See new changes below (includes three commits, with final commit the working version). I got it to work for pulling in one instance of nav/footer. What I want to know is: What if we have separated nav snippets manually in a template, if we pulled in all nav snippets, wouldn't that override it? How can we pull in only current navs/footers and not future navs/footers if this migration is run again. That's why I didn't do .all() for it and just did the first navbar or footer. Hopefully I am making sense.

I tested it by creating a new db in dev branch, adding navbars and footers the usual way, then switching to my branch and migrating.  @vsalvino

Reply:
Also, new tests will need to be written because this fails pytest really bad....

Reply:
Maybe this part isn't necessary? 
`layout.site_navbar = []
 layout.save()`
In case it's not the tests that need updated....